spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/MqttPahoMessageHandler.java

```diff
@@ -208,7 +204,7 @@ private void sendDeliveryComplete(IMqttDeliveryToken token) {
 	}
 
 	@Override
-	public void connectionLost(Throwable cause) {
+	public synchronized void connectionLost(Throwable cause) {
```

Comment:
`checkConnection()` sets it to not-null at line 171 and returns it.

If this method is not synchronized, it could set it null again between lines 171 and 180.

Also, we don't want this to be called while the other thread is processing the `if` at line 158 - it could cause an NPE on line 158 or 159.

We could restructure that method to use a local variable but that would require an unnecessary assignment in the happy path.

This will be called rarely so I don't think synchronizing it is a big deal.